src/Microsoft.AspNet.Mvc.Razor/RazorViewEngine.cs

```diff
@@ -233,14 +235,43 @@ public virtual IEnumerable<string> AreaViewLocationFormats
             string pageName,
             bool isPartial)
         {
-            if (IsApplicationRelativePath(pageName))
+            var applicationRelativePath = pageName;
+            if (!IsApplicationRelativePath(applicationRelativePath) && IsRelativePath(applicationRelativePath))
             {
-                var applicationRelativePath = pageName;
-                if (!pageName.EndsWith(ViewExtension, StringComparison.OrdinalIgnoreCase))
+                // Given a relative path that is not yet app-relative.
+                // Interpret given path relative to currently-executing view, if any.
+                var executingPath = (context as ViewContext)?.ExecutingFilePath;
```

Comment:
Yes, this is meant to support relative paths for partials, top-level views, top-level views for view components, ...